lib/auth/permissions.go

```diff
@@ -181,6 +181,38 @@ func GetCheckerForBuiltinRole(role teleport.Role) (services.AccessChecker, error
 				},
 			})
 	case teleport.RoleProxy:
+		if clusterConfig.GetSessionRecording() == services.RecordAtProxy {
```

Comment:
the previous version was just adding a couple of rules, why did you change it to the whole new role?